From 9656cb372ef56dd53d0365c85c29dc0ee812e40f Mon Sep 17 00:00:00 2001
From: Charles Ci-Jyun Wu <dminus@andestech.com>
Date: Thu, 28 Sep 2023 16:42:18 +0800
Subject: [PATCH 013/168] riscv: dts: andes: add dma-noncoherent property to
 support noncoherent device

Under CONFIG_RISCV, which selects ARCH_DMA_DEFAULT_COHERENT, all devices
are assumed to be DMA-coherent by default.

commit c00a60d6f4a14 ("of: address: always use dma_default_coherent
for default coherency")

For noncoherent devices that require noncoherent DMA, 'dma-noncoherent'
should be added to their device nodes.

Signed-off-by: Charles Ci-Jyun Wu <dminus@andestech.com>
---
 arch/riscv/boot/dts/andes/a25_c1_d_dsp_noncoherent_ae350.dts | 5 +++++
 .../boot/dts/andes/a27l2_c1_d_dsp_noncoherent_ae350.dts      | 5 +++++
 arch/riscv/boot/dts/andes/a45_c1_d_dsp_noncoherent_ae350.dts | 5 +++++
 .../riscv/boot/dts/andes/ax25_c1_d_dsp_noncoherent_ae350.dts | 5 +++++
 .../boot/dts/andes/ax27l2_c1_d_dsp_noncoherent_ae350.dts     | 5 +++++
 .../riscv/boot/dts/andes/ax45_c1_d_dsp_noncoherent_ae350.dts | 5 +++++
 6 files changed, 30 insertions(+)

diff --git a/arch/riscv/boot/dts/andes/a25_c1_d_dsp_noncoherent_ae350.dts b/arch/riscv/boot/dts/andes/a25_c1_d_dsp_noncoherent_ae350.dts
index 7a99528e0c1f..f6f519835021 100644
--- a/arch/riscv/boot/dts/andes/a25_c1_d_dsp_noncoherent_ae350.dts
+++ b/arch/riscv/boot/dts/andes/a25_c1_d_dsp_noncoherent_ae350.dts
@@ -172,6 +172,7 @@ mac@e0100000 {
 			reg = <0x00 0xe0100000 0x00 0x1000>;
 			interrupts = <0x13 0x04>;
 			interrupt-parent = <0x02>;
+			dma-noncoherent;
 		};
 
 		smu@f0100000 {
@@ -189,6 +190,7 @@ mmc@f0e00000 {
 			fifo-depth = <0x10>;
 			dma-names = "rxtx";
 			dmas = <0x03 0x09>;
+			dma-noncoherent;
 		};
 
 		dma@f0c00000 {
@@ -199,6 +201,7 @@ dma@f0c00000 {
 			dma-channels = <0x08>;
 			#dma-cells = <0x01>;
 			phandle = <0x03>;
+			dma-noncoherent;
 		};
 
 		lcd@e0200000 {
@@ -206,6 +209,7 @@ lcd@e0200000 {
 			reg = <0x00 0xe0200000 0x00 0x1000>;
 			interrupts = <0x14 0x04>;
 			interrupt-parent = <0x02>;
+			dma-noncoherent;
 		};
 
 		smc@e0400000 {
@@ -225,6 +229,7 @@ snd@f0d00000 {
 			reg = <0x00 0xf0d00000 0x00 0x1000>;
 			interrupts = <0x11 0x04>;
 			interrupt-parent = <0x02>;
+			dma-noncoherent;
 		};
 
 		pmu {
diff --git a/arch/riscv/boot/dts/andes/a27l2_c1_d_dsp_noncoherent_ae350.dts b/arch/riscv/boot/dts/andes/a27l2_c1_d_dsp_noncoherent_ae350.dts
index 3b3c4d902851..191e9684223d 100644
--- a/arch/riscv/boot/dts/andes/a27l2_c1_d_dsp_noncoherent_ae350.dts
+++ b/arch/riscv/boot/dts/andes/a27l2_c1_d_dsp_noncoherent_ae350.dts
@@ -185,6 +185,7 @@ mac@e0100000 {
 			reg = <0x00 0xe0100000 0x00 0x1000>;
 			interrupts = <0x13 0x04>;
 			interrupt-parent = <0x03>;
+			dma-noncoherent;
 		};
 
 		smu@f0100000 {
@@ -202,6 +203,7 @@ mmc@f0e00000 {
 			fifo-depth = <0x10>;
 			dma-names = "rxtx";
 			dmas = <0x03 0x09>;
+			dma-noncoherent;
 		};
 
 		dma@f0c00000 {
@@ -212,6 +214,7 @@ dma@f0c00000 {
 			dma-channels = <0x08>;
 			#dma-cells = <0x01>;
 			phandle = <0x05>;
+			dma-noncoherent;
 		};
 
 		lcd@e0200000 {
@@ -219,6 +222,7 @@ lcd@e0200000 {
 			reg = <0x00 0xe0200000 0x00 0x1000>;
 			interrupts = <0x14 0x04>;
 			interrupt-parent = <0x03>;
+			dma-noncoherent;
 		};
 
 		smc@e0400000 {
@@ -238,6 +242,7 @@ snd@f0d00000 {
 			reg = <0x00 0xf0d00000 0x00 0x1000>;
 			interrupts = <0x11 0x04>;
 			interrupt-parent = <0x03>;
+			dma-noncoherent;
 		};
 
 		pmu {
diff --git a/arch/riscv/boot/dts/andes/a45_c1_d_dsp_noncoherent_ae350.dts b/arch/riscv/boot/dts/andes/a45_c1_d_dsp_noncoherent_ae350.dts
index 9ab171cc8b70..32a6b05fae70 100644
--- a/arch/riscv/boot/dts/andes/a45_c1_d_dsp_noncoherent_ae350.dts
+++ b/arch/riscv/boot/dts/andes/a45_c1_d_dsp_noncoherent_ae350.dts
@@ -174,6 +174,7 @@ mac@e0100000 {
 			reg = <0x00 0xe0100000 0x00 0x1000>;
 			interrupts = <0x13 0x04>;
 			interrupt-parent = <0x02>;
+			dma-noncoherent;
 		};
 
 		smu@f0100000 {
@@ -191,6 +192,7 @@ mmc@f0e00000 {
 			fifo-depth = <0x10>;
 			dma-names = "rxtx";
 			dmas = <0x03 0x09>;
+			dma-noncoherent;
 		};
 
 		dma@f0c00000 {
@@ -201,6 +203,7 @@ dma@f0c00000 {
 			dma-channels = <0x08>;
 			#dma-cells = <0x01>;
 			phandle = <0x03>;
+			dma-noncoherent;
 		};
 
 		lcd@e0200000 {
@@ -208,6 +211,7 @@ lcd@e0200000 {
 			reg = <0x00 0xe0200000 0x00 0x1000>;
 			interrupts = <0x14 0x04>;
 			interrupt-parent = <0x02>;
+			dma-noncoherent;
 		};
 
 		smc@e0400000 {
@@ -227,6 +231,7 @@ snd@f0d00000 {
 			reg = <0x00 0xf0d00000 0x00 0x1000>;
 			interrupts = <0x11 0x04>;
 			interrupt-parent = <0x02>;
+			dma-noncoherent;
 		};
 
 		pmu {
diff --git a/arch/riscv/boot/dts/andes/ax25_c1_d_dsp_noncoherent_ae350.dts b/arch/riscv/boot/dts/andes/ax25_c1_d_dsp_noncoherent_ae350.dts
index 69d7b4db865e..bfa83ed86b82 100644
--- a/arch/riscv/boot/dts/andes/ax25_c1_d_dsp_noncoherent_ae350.dts
+++ b/arch/riscv/boot/dts/andes/ax25_c1_d_dsp_noncoherent_ae350.dts
@@ -172,6 +172,7 @@ mac@e0100000 {
 			reg = <0x00 0xe0100000 0x00 0x1000>;
 			interrupts = <0x13 0x04>;
 			interrupt-parent = <0x02>;
+			dma-noncoherent;
 		};
 
 		smu@f0100000 {
@@ -189,6 +190,7 @@ mmc@f0e00000 {
 			fifo-depth = <0x10>;
 			dma-names = "rxtx";
 			dmas = <0x03 0x09>;
+			dma-noncoherent;
 		};
 
 		dma@f0c00000 {
@@ -199,6 +201,7 @@ dma@f0c00000 {
 			dma-channels = <0x08>;
 			#dma-cells = <0x01>;
 			phandle = <0x03>;
+			dma-noncoherent;
 		};
 
 		lcd@e0200000 {
@@ -206,6 +209,7 @@ lcd@e0200000 {
 			reg = <0x00 0xe0200000 0x00 0x1000>;
 			interrupts = <0x14 0x04>;
 			interrupt-parent = <0x02>;
+			dma-noncoherent;
 		};
 
 		smc@e0400000 {
@@ -225,6 +229,7 @@ snd@f0d00000 {
 			reg = <0x00 0xf0d00000 0x00 0x1000>;
 			interrupts = <0x11 0x04>;
 			interrupt-parent = <0x02>;
+			dma-noncoherent;
 		};
 
 		pmu {
diff --git a/arch/riscv/boot/dts/andes/ax27l2_c1_d_dsp_noncoherent_ae350.dts b/arch/riscv/boot/dts/andes/ax27l2_c1_d_dsp_noncoherent_ae350.dts
index 31709f8a73a8..ce362b931189 100644
--- a/arch/riscv/boot/dts/andes/ax27l2_c1_d_dsp_noncoherent_ae350.dts
+++ b/arch/riscv/boot/dts/andes/ax27l2_c1_d_dsp_noncoherent_ae350.dts
@@ -185,6 +185,7 @@ mac@e0100000 {
 			reg = <0x00 0xe0100000 0x00 0x1000>;
 			interrupts = <0x13 0x04>;
 			interrupt-parent = <0x03>;
+			dma-noncoherent;
 		};
 
 		smu@f0100000 {
@@ -202,6 +203,7 @@ mmc@f0e00000 {
 			fifo-depth = <0x10>;
 			dma-names = "rxtx";
 			dmas = <0x03 0x09>;
+			dma-noncoherent;
 		};
 
 		dma@f0c00000 {
@@ -212,6 +214,7 @@ dma@f0c00000 {
 			dma-channels = <0x08>;
 			#dma-cells = <0x01>;
 			phandle = <0x05>;
+			dma-noncoherent;
 		};
 
 		lcd@e0200000 {
@@ -219,6 +222,7 @@ lcd@e0200000 {
 			reg = <0x00 0xe0200000 0x00 0x1000>;
 			interrupts = <0x14 0x04>;
 			interrupt-parent = <0x03>;
+			dma-noncoherent;
 		};
 
 		smc@e0400000 {
@@ -238,6 +242,7 @@ snd@f0d00000 {
 			reg = <0x00 0xf0d00000 0x00 0x1000>;
 			interrupts = <0x11 0x04>;
 			interrupt-parent = <0x03>;
+			dma-noncoherent;
 		};
 
 		pmu {
diff --git a/arch/riscv/boot/dts/andes/ax45_c1_d_dsp_noncoherent_ae350.dts b/arch/riscv/boot/dts/andes/ax45_c1_d_dsp_noncoherent_ae350.dts
index 04284624c129..bb872e694dc4 100644
--- a/arch/riscv/boot/dts/andes/ax45_c1_d_dsp_noncoherent_ae350.dts
+++ b/arch/riscv/boot/dts/andes/ax45_c1_d_dsp_noncoherent_ae350.dts
@@ -174,6 +174,7 @@ mac@e0100000 {
 			reg = <0x00 0xe0100000 0x00 0x1000>;
 			interrupts = <0x13 0x04>;
 			interrupt-parent = <0x02>;
+			dma-noncoherent;
 		};
 
 		smu@f0100000 {
@@ -191,6 +192,7 @@ mmc@f0e00000 {
 			fifo-depth = <0x10>;
 			dma-names = "rxtx";
 			dmas = <0x03 0x09>;
+			dma-noncoherent;
 		};
 
 		dma@f0c00000 {
@@ -201,6 +203,7 @@ dma@f0c00000 {
 			dma-channels = <0x08>;
 			#dma-cells = <0x01>;
 			phandle = <0x03>;
+			dma-noncoherent;
 		};
 
 		lcd@e0200000 {
@@ -208,6 +211,7 @@ lcd@e0200000 {
 			reg = <0x00 0xe0200000 0x00 0x1000>;
 			interrupts = <0x14 0x04>;
 			interrupt-parent = <0x02>;
+			dma-noncoherent;
 		};
 
 		smc@e0400000 {
@@ -227,6 +231,7 @@ snd@f0d00000 {
 			reg = <0x00 0xf0d00000 0x00 0x1000>;
 			interrupts = <0x11 0x04>;
 			interrupt-parent = <0x02>;
+			dma-noncoherent;
 		};
 
 		pmu {
-- 
2.34.1

